proposals/restructure-KF-docs-proposal.md

```diff
@@ -0,0 +1,226 @@
+    - Kubeflow on Linux
+    - Kubeflow on macOS
+    - Kubeflow on Windows
+    - MiniKF
+    - Deploy using MiniKF on GCP
+    - Deploying with minikube on a single node
+    - Kubeflow on MicroK8s
+- **Components of Kubeflow**
+  - Central Dashboard
+    - Central Dashboard 
+    - Registration Flow
+  - Metadata
+  - Jupyter Notebooks
+  - Fairing 
+    - Overview of Kubeflow Fairing
+    - Install Kubeflow Fairing
+    - Configure Kubeflow Fairing
+    - Fairing on Azure
+    - Fairing on GCP
+    - Tutorials
+    - Reference
+  - Feature Store
+    - Introduction to Feast
+    - Getting started with Feast
+  - Frameworks for Training
+    - Chainer Training
+    - MPI Training
+    - MXNet Training
+    - PyTorch Training
+    - TensorFlow Training (TFJob)
+  - Hyperparameter Tuning
+    - Introduction to Katib
+    - Getting started with Katib
+    - Running an experiment
+    - Katib Configuration Overview
+    - Environment Variables for Katib Components
+  - Pipelines
+    - Pipelines
+  - Tools for Serving
+    - Overview
+    - KFServing
+    - Seldon Core Serving
+    - BentoML
+    - NVIDIA Triton Inference Server
+    - TensorFlow Serving
+    - TensorFlow Batch Prediction
+  - Multi-Tenancy in Kubeflow
+    - Introduction to Multi-user Isolation
+    - Design for Multi-user Isolation
+    - Getting Started with Multi-user Isolation
+  - Miscellaneous
+    - Nuclio functions
+- **Jupyter Notebooks**
+  - Overview of Jupyter Notebooks in Kubeflow
+  - Set Up Your Notebooks
+  - Create a Custom Jupyter Image
+  - Submit Kubernetes Resources
+  - Build a Docker Image on GCP
+  - Troubleshooting Guide
+- **Pipelines**
+  - Pipelines Quickstart
+  - Installing Pipelines
+    - Installation Options for Kubeflow Pipelines
+    - Kubeflow Pipelines Standalone Deployment
+    - Deploying Kubeflow Pipelines on a local cluster
+  - Understanding Pipelines
+    - Overview of Kubeflow Pipelines
+    - Introduction to the Pipelines Interfaces
+    - Concepts
+  - Building Pipelines with the SDK
+    - Introduction to the Pipelines SDK
+    - Install the Kubeflow Pipelines SDK
+    - Build Components and Pipelines
+    - Create Reusable Components
+    - Building Python function-based components
+    - Best Practices for Designing Components
+    - Pipeline Parameters
+    - Python Based Visualizations
+    - Visualize Results in the Pipelines UI
+    - Pipeline Metrics
+    - DSL Static Type Checking
+    - DSL Recursion
+    - Using environment variables in pipelines
+    - GCP-specific Uses of the SDK
+    - Manipulate Kubernetes Resources as Part of a Pipeline
+  - Multi-user Isolation for Pipelines
+  - Caching
+  - Upgrading
+  - Samples and Tutorials
+    - Using the Kubeflow Pipelines Benchmark Scripts
+    - Using the Kubeflow Pipelines SDK
+    - Experiment with the Kubeflow Pipelines API
+    - Experiment with the Pipelines Samples
+    - Run a Cloud-specific Pipelines Tutorial
+  - Troubleshooting
+  - Reference
```

Comment:
Pipelines has their own Reference.
Does it make sense to move Pipelines reference under /docs/reference where other projects are located ?

Reply:
It seems that Pipelines and Fairing are the only components that have references outside of reference. Might make sense to merge everything.

Reply:
I think this comes back to the question of splitting component applications onto separate sites. In that case, each component application's documentation will need to be self-contained.

Reply:
From the conversations in the community meetings that I have attended, it sounds like we are moving towards separate sites for each component. Would it make sense to add a separate file for modeling out what that restructure would look like if we split the component docs onto other sites? Or, should we model that in this doc?

Reply:
As for Reference, if we move to the separate domains method, I would argue that we should keep a copy of reference on both, in the same manner as we currently do with `manifests` in the github org. 

@joeliedtke happy with either. I guess none of this is set in stone. With this document I aim at identifying the quickest structural changes we can make to make the website more clear. We still need to rely on `WG-leads` to make sure their lower-level docs structure makes sense.

Reply:
It looks like I can't edit files in this PR, so here is a quick sketch of what I have been thinking about for information architecture of Kubeflow.org after the component docs move to their own site:

- Kubeflow Home Page
- Getting started (same as current proposal, consider renaming to About Kubeflow)
- Community (same as current proposal)
- Components
  - Jupyter Notebooks (Single page that describes what this component is and where to learn more)
  - Central Dashboard (Single page that describes what this component is and where to learn more)
  - Metadata (Single page that describes what this component is and where to learn more)
  - Fairing (Single page that describes what this component is and where to learn more)
  - Feature Store (Single page that describes what this component is and where to learn more)
  - Frameworks for training (Single page that describes what this component is and where to learn more)
  - Hyperparameter Tuning (Single page that describes what this component is and where to learn more)
  - Kubeflow Pipelines (Single page that describes what this component is and where to learn more)
  - Jupyter Notebooks (Single page that describes what this component is and where to learn more)
  - Tools for Serving (Single page that describes what this component is and where to learn more)
  - Multi-Tenancy (Single page that describes what this component is and where to learn more)
  - Nuclio functions (Single page that describes what this component is and where to learn more)
- Docs
  - Deployment (same as current proposal)
  - Configuring Kubeflow (same as Setups in the current proposal)
  - Resources (Review content, if it is general to Kubeflow then keep this section)
  - Troubleshooting
  - Reference (Review content, if it is general to Kubeflow then keep this section)

The thought is to focus the Kubeflow.org site on describing the Kubeflow project, components, and deployment options. All of the docs for the component applications will be hosted on other sites.

Reply:
Hi @joeliedtke, thank you for the feedback!

Regarding whether docs are hosted on other sites, I think the downsides might be consistency, driving people away from Kubeflow.org and harder to make sure the websites are high quality. But despite that, I'm happy with what the community decides.

As per your suggestions:
- I like the `Kubeflow Home Page` on the sidebar. 
- I think `Docs` is not a good word there as all are docs, but I see the point of merging those 5 things under an umbrella.  
- I also like `Configuring` instead of `Setups`